pkg/defaults/defaults.go

```diff
@@ -177,6 +178,19 @@ func FromConfig(
 				if test.ClusterProfile != "" {
 					step = steps.LeaseStep(leaseClient, test.ClusterProfile.LeaseType(), step, jobSpec.Namespace, namespaceClient)
 				}
+				for _, subStep := range append(append(test.Pre, test.Test...), test.Post...) {
+					if link, ok := subStep.FromImageTag(); ok {
+						config := api.InputImageTagStepConfiguration{
+							BaseImage: *subStep.FromImage,
+							To:        link,
+						}
+						srcClient, err := clusterImageStreamClient(imageClient, clusterConfig, config.BaseImage.Cluster, kubeconfigs)
+						if err != nil {
+							return nil, nil, fmt.Errorf("unable to access image stream tag on remote cluster: %v", err)
+						}
+						additional = append(additional, steps.InputImageTagStep(config, srcClient, imageClient, jobSpec, dryLogger))
```

Comment:
Does the execution graph handle duplicates or will we start two independent steps?

Reply:
They will just race to create the object, as is the case already between two different jobs that execute the same steps, so it should not be an issue.

Reply:
I don't think it should break anything, too. I think it's the first time we let things race this way in a single ci-op run, though, so it will be worth keeping the eye on.